instrumentation/github.com/astaxie/beego/README.md

```diff
@@ -0,0 +1,82 @@
+## `go.opentelemetry.io/contrib/instrumentation/github.com/astaxie/beego`
+
+This package provides tracing and metrics to the `github.com/astaxie/beego` package.
+
+To enable tracing and metrics in your beego application:
+
+```go
+package main
+
+import (
+        "github.com/astaxie/beego"
+
+        otelBeego "go.opentelemetry.io/contrib/instrumentation/github.com/astaxie/beego"
+)
+
+func main() {
+        // Init the tracer and metric export pipelines
+
+        // Create your routes
+        
+        // Create the MiddleWare
+        mware := otelBeego.NewOTelBeegoMiddleWare("example-server")
+
+        // Run the server with the MiddleWare
+        beego.RunWithMiddleWares("localhost:8080", mware)
+}
+```
+
+You can customize instrumentation by passing any of the following options to `NewOtelBeegoMiddleWare`:
+
+| Function | Description |
+| -------- | ----------- |
+| `WithTracer(trace.Tracer)` | The tracer to be used to create spans for the beego server. If not specified, `global.Tracer("go.opentelemetry.io/contrib/instrumentation/github.com/astaxie/beego")` will be used. |
+| `WithMeter(metric.Meter)` | The meter to be used to create the instruments. If not specified, `global.Meter("go.opentelemery.io/contrib/instrumentation/github.com/astaxie/beego")` will be used. |
+| `WithPropagators(propagation.Propagators)` | The propagators to be used. If not specified, `global.Propagators()` will be used. |
```

Comment:
This will need to be included in this cleanup: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/213

Reply:
Yea definitely. Did you want this changed now? 
I could remove the options and replace them with `With*Provider()` options?

Reply:
Yeah, if you can that would be awesome! :heart: